### PR TITLE
81 updating product inventory and purchase for requried fields

### PR DESCRIPTION
**Pull request by SruthiKonda12**
updated mandatory fields for product, inventory and purchase modules.

### CODE REVIEW

File: apps/products/models.py [hunk 1]
```diff
@@ -187,48 +181,48 @@ def products_picture(instance, filename):
...
     brand_id = models.ForeignKey(ProductBrands, on_delete=models.CASCADE, null=True, default=None, db_column = 'brand_id')
-    purchase_warranty_months = models.IntegerField()
-    sales_warranty_months = models.IntegerField()
+    purchase_warranty_months = models.IntegerField(null=True, default=None)
+    sales_warranty_months = models.IntegerField(null=True, default=None)
+    STATUS_CHOICES = [
+        ('Active', 'Active'),
+        ('Inactive', 'Inactive'),
+        ('Pending','Pending'),
+    ]
```
Comment (thulasiram1216) [inline apps/products/models.py:225]: status ENUM('Active', 'Inactive'), correct it.

File: apps/products/models.py [hunk 2]
```diff
@@ -187,48 +181,48 @@ def products_picture(instance, filename):
...
     status = models.CharField(max_length=10, choices=STATUS_CHOICES, default='Inclusive')
```
Comment (thulasiram1216) [inline apps/products/models.py:226]: Why default='Inclusive' for STATUS_CHOICES, please remove.
There's no default for this field

File: apps/purchase/models.py [hunk 3]
```diff
@@ -3,27 +3,27 @@
...
+    order_date = models.DateField()
+    order_no = models.CharField(max_length=255)
     ref_no = models.CharField(max_length=255, null=True, default=None)
-    ref_date = models.DateField(null=True, default=None)
+    ref_date = models.DateField()
     vendor_agent_id = models.ForeignKey(VendorAgent, on_delete=models.CASCADE, null=True, default=None, db_column = 'vendor_agent_id')
     TAX_CHOICES = [('inclusive', 'Inclusive'),
                    ('exclusive', 'Exclusive')
                 ]
     tax = models.CharField(max_length=20, choices=TAX_CHOICES , default= 'Inclusive')
```
Comment (thulasiram1216) [inline apps/purchase/models.py:24]: Remove default here

File: apps/masters/models.py [hunk 4]
```diff
@@ -232,15 +232,15 @@ def __str__(self):
...
     RATE_ON_CHOICES = [ 
         ('Qty', 'Qty'),
         ('Amount', 'Amount'),
         ('Both','Both'),
     ]
-    brand_salesman_id = models.AutoField(primary_key=True)
-    code = models.CharField(max_length=50)
-    name = models.CharField(max_length=255)
-    commission_rate = models.DecimalField(max_digits=18, decimal_places=2)
     rate_on = models.CharField(max_length=10, choices=RATE_ON_CHOICES, blank=True, null=True)
```
[the remark this answers is not in the document]
Comment (SruthiKonda12) [reply apps/masters/models.py:244]: Done

File: apps/products/models.py [hunk 5]
```diff
@@ -1,12 +1,10 @@
...
+import os,uuid
 from django.db import models
-from apps.masters.models import *
 from django.dispatch import receiver
 from django.db.models.signals import pre_delete
-from utils_methods import EncryptedTextField
+from utils_methods import *
 from utils_variables import *
-
+from apps.masters.models import *
```
[the remark this answers is not in the document]
Comment (SruthiKonda12) [reply apps/products/models.py:7]: Done

File: apps/products/models.py (hunk 1 above)
Comment (SruthiKonda12) [reply apps/products/models.py:225]: changed

File: apps/products/models.py (hunk 2 above)
Comment (SruthiKonda12) [reply apps/products/models.py:226]: removed

File: apps/purchase/models.py (hunk 3 above)
Comment (SruthiKonda12) [reply apps/purchase/models.py:24]: solved

**Pull request merged**